testsuite/unittests/test_gap_junction.sli

```diff
@@ -204,79 +204,93 @@ M_ERROR setverbosity
-        >> ConnectLayers
-} pass_or_die
-
-% Check that we can use gap junction with source-driven connect in topology
-{
-  ResetKernel
-  /lr << /rows 2
-         /columns 3
-         /elements /hh_psc_alpha_gap      
-      >> CreateLayer def
-  
-  lr lr << /connection_type (divergent)
-           /synapse_model /gap_junction 
-           /make_symmetric true
-           /weights 0.5
-        >> ConnectLayers
-} pass_or_die
-
-% Check that we can use gap junction with convergent connect in topology
-{
-  ResetKernel
-  /lr << /rows 2
-         /columns 3
-         /elements /hh_psc_alpha_gap      
-      >> CreateLayer def
-  
-  lr lr << /connection_type (convergent)
-           /synapse_model /gap_junction 
-           /make_symmetric true
-           /weights 0.5
-           /number_of_connections 3
-        >> ConnectLayers
-} pass_or_die
-
-
-% Check that we can use gap junction with divergent connect in topology
-{
-  ResetKernel
-  /lr << /rows 2
-         /columns 3
-         /elements /hh_psc_alpha_gap      
-      >> CreateLayer def
+  /tgt_driven << /connection_type (convergent)
+                 /synapse_model /gap_junction
+                 /make_symmetric true
+                 /weights << /uniform << /min 0.3 /max 0.6 >> >>
+              >> def
+  /src_driven << /connection_type (divergent)
+                 /synapse_model /gap_junction
+                 /make_symmetric true
+                 /weights << /uniform << /min 0.3 /max 0.6 >> >>
+              >> def
+  /convergent << /connection_type (convergent)
+                 /synapse_model /gap_junction
+                 /make_symmetric true
+                 /weights << /uniform << /min 0.3 /max 0.6 >> >>
+                 /number_of_connections 3
+              >> def
+  /divergent << /connection_type (divergent)
+                /synapse_model /gap_junction
+                /make_symmetric true
+                /weights << /uniform << /min 0.3 /max 0.6 >> >>
+                /number_of_connections 3
+             >> def
   
-  lr lr << /connection_type (divergent)
-           /synapse_model /gap_junction 
-           /make_symmetric true
-           /weights 0.5
-           /number_of_connections 3
-        >> ConnectLayers
-} pass_or_die
+  true
+  [ tgt_driven src_driven convergent divergent  ]
+  {
+    /connspec Set
+    ResetKernel
+    
+    /src_lr << /rows 2
+           /columns 3
+           /elements /hh_psc_alpha_gap
+        >> CreateLayer def
+    /tgt_lr << /rows 2
+           /columns 3
+           /elements /hh_psc_alpha_gap
+        >> CreateLayer def
+    
+    src_lr tgt_lr connspec ConnectLayers
+    
+    % Storing connection info to array.
+    /nodes src_lr GetLocalLeaves def
+    << /source nodes >> GetConnections
+    dup length exch 
+    {
+      dup /source get exch
+      dup /target get exch
+      dup /synapse_model get exch
+      /weight get
+      4 arraystore exch
+    } forall
+    arraystore
+    /source_array Set
```

Comment:
You can simplify the code here quite a bit using `Map`, and I think you do not need to select by `/source`:
```
<< >> GetConnections 
{ /conn Set
  [ /source /target /synapse_model /weight ] { conn exch get } Map
} Map
```
and the same thing again with the position of source and target swapped.